src/api/api.ts

```diff
@@ -1 +1,53 @@
-export const BASE_URL = 'https://mate.academy/students-api';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+const BASE_URL = 'https://mate.academy/students-api';
+
+type CommentInfo = Omit<Comment, 'id'>;
+
+function getData<T>(url: string): Promise<T> {
+  return fetch(`${BASE_URL}/${url}`)
+    .then(response => response.json());
+}
+
+export const getAllPosts = async () => {
+  const posts = await getData<Post[]>('posts');
+
+  return posts;
+};
+
+export const getUserPosts = async (userId: number) => {
+  const posts = await getAllPosts();
+
+  if (userId === 0) {
+    return posts;
+  }
+
+  return posts.filter((post => post.userId === userId));
+};
```

Comment:
you can get access to the user posts by adding /posts?userId=${userId} to the url